kyuubi-common/src/main/scala/org/apache/kyuubi/util/KyuubiHadoopUtils.scala

```diff
@@ -42,8 +42,16 @@ object KyuubiHadoopUtils {
 
   def newHadoopConf(
       conf: KyuubiConf,
-      loadDefaults: Boolean = true): Configuration = {
+      loadDefaults: Boolean = true,
+      engineEnv: Boolean = false): Configuration = {
     val hadoopConf = new Configuration(loadDefaults)
+    if (engineEnv) {
+      conf.getEnvs.get("HADOOP_CONF_DIR").foreach { hadoopConfDir =>
```

Comment:
Great, I saw this issue too, waiting for you to merge to master 😄

Reply:
it's overkilling to me, do you actually need a new YarnConfiguation instead?

Reply:
I tried to use `Configuration.addDefaultResource` but failed.

Reply:
> If you use YarnConfiguation, I guess it could trigger reload yarn-site.xml

How can I specify the yarn-site.xml?

Reply:
thanks